inc/class-customizer-guided-tour.php

```diff
@@ -0,0 +1,232 @@
+			$current_theme = wp_get_theme();
+
+			if ( 'Storefront' == $current_theme->get( 'Name' ) ) {
+				$redirect_url = add_query_arg( array(
+					'sf_guided_tour' => '1',
+					'sf_tasks' => 'homepage',
+					'theme' => $_GET['theme'],
+				), admin_url( '/customize.php' ) );
+
+				wp_redirect( $redirect_url );
+				exit;
+			}
+		}
+
+		/**
+		 * Customizer enqueues.
+		 *
+		 * @since 0.8.7
+		 */
+		public function customize_scripts() {
+			global $storefront_version;
+
+			wp_enqueue_style( 'wc-api-dev-guided-tour', WC_API_Dev::$plugin_url . 'assets/css/admin/customizer.css', array(), $storefront_version, 'all' );
+
+			wp_enqueue_script( 'wc-api-dev-guided-tour', WC_API_Dev::$plugin_url . 'assets/js/admin/customizer.js', array( 'jquery', 'wp-backbone' ), $storefront_version, true );
+
+			wp_localize_script( 'wc-api-dev-guided-tour', '_wpCustomizeWcApiDevGuidedTourSteps', $this->guided_tour_steps() );
+			wp_localize_script( 'wc-api-dev-guided-tour', '_wpCustomizeWcApiDevGuidedSettings', $this->guided_tour_settings() );
+		}
+
+		/**
+		 * Template for steps.
+		 *
+		 * @since 0.8.7
+		 */
+		public function print_templates() {
+			?>
+			<script type="text/html" id="tmpl-wc-api-guided-tour-step">
+				<div class="sf-guided-tour-step">
+					<a class="sf-guided-tour-close" href="#">
+						<span class="dashicons dashicons-no-alt"></span>
+					</a>
+					<# if ( data.title ) { #>
+						<h2>{{ data.title }}</h2>
+					<# } #>
+					{{{ data.message }}}
+					<# if ( data.altStep ) { #>
+						<a class="sf-nux-button sf-nux-alt-button {{ data.altStep.className }}" href="#">
+							{{ data.altStep.buttonText }}
+						</a>
+					<# } #>
+					<# if ( data.buttonText ) { #>
+						<a class="sf-nux-button sf-nux-primary-button {{ data.className }}" href="#">
+							{{ data.buttonText }}
+						</a>
+					<# } #>
+
+					<# if ( data.showSkip ) { #>
+						<a class="sf-guided-tour-skip" href="#">
+							<?php esc_attr_e( 'Skip this step', 'wc-api-dev' ); ?>
+						</a>
+					<# } #>
+				</div>
+			</script>
+			<?php
+		}
+
+		/**
+		 * Guided tour settings.
+		 *
+		 * @since 0.8.7
+		 */
+		public function guided_tour_settings() {
+			$show_tour = isset( $_GET['store-wpcom-nux'] );
+			$theme_supports_woo = current_theme_supports( 'woocommerce' );
+
+			return array(
+				'autoStartTour' => ( bool ) $show_tour,
+				'showTourAlert' => ( bool ) ! $show_tour,
+				'alertMessage'  => __( 'It looks like your current theme isn\'t ready for shop features yet - your shop and product pages might look a little funny.<br/><br/>We reccomend switching themes to Storefront. <a href="/wp-admin/customize.php?theme=storefront&sf_guided_tour=1&sf_tasks=homepage">Click here</a> to get started.', 'storefront' ),
+			);
+		}
+
+		/**
+		 * Guided tour steps.
+		 *
+		 * @since 0.8.7
+		 */
+		public function guided_tour_steps() {
+			$steps = array();
+
+			$steps[] = array(
+				'title'       => __( 'Customize Your Store', 'storefront' ),
```

Comment:
Question: I'm assuming these strings are already translated by Storefront and that's why we have the textdomain here?

Reply:
Yes, a big part of the logic in this PR was copied over from the current storefront tour, the translation domain being one of those items. Not sure the best thing to do here, but perhaps when we split out wc-api-dev into two plugins, we can properly handle translation/pot flow/domains then?

Reply:
Yeah I think that makes sense. We've done something similar for our other strings that might need to be handled when we split plugins. I just wanted to double check.